We.Sell.Bread.API.Unit.Tests/Tests/Services/ProductServiceTests.cs

```diff
@@ -19,7 +19,46 @@ public void GivenDataExistWhenRetrievingProductReturnTypeMustBeListOfProductDeta
 
             product.Should().NotBeNull();
             product.Should().BeOfType(typeof(List<ProductDetailsDto>));
-           
+
+        }
+
+        [Fact]
+        public void GivenIdShouldExistAndReturnTypeObjectProductDetailsDto()
+        {
+            ProductDetailsDto? product = _productService.GetProduct(new Guid("411fbad3-925e-4044-9269-f962641f9277"));
+            string realName = "Pumpernickel Bread";
```

Comment:
Just one thing to add from my side:

As a best practice for writing tests (and code), you do not want to hard code your assignments as this opens up your data to human error and can affect the integrity of our tests. It also means you have to manually create more test data as you add more tests and that gets tedious.

For a more automated and consistent way  of handling test data, make use of libraries that were specifically made for this purpose, unless we need to use real data for our test cases. For this project, we use a library called Faker, that has many convenient methods for creating test data. See the CustomerServiceTests for an example. You can find the list of Faker methods here: https://github.com/Kuree/Faker.Net. To help you a bit, Hacker class has methods for generic verbs, nouns and descriptions.